boa/vm/fork.py

```diff
@@ -29,11 +29,10 @@
 _EMPTY = b""  # empty rlp stuff
 
 
-class CachingRPC(EthereumRPC):
-    def __init__(self, url: str, cache_file: str = DEFAULT_CACHE_DIR):
-        super().__init__(url)
-
+class CachingRPC(RPC):
+    def __init__(self, rpc: RPC, cache_file: str = DEFAULT_CACHE_DIR):
```

Comment:
i think we should override `CachingRPC.__new__()` so that if it receives a `CachingRPC`, it just returns the argument.

Reply:
this way in AccountDBFork, we can always wrap whatever rpc we receive in `CachingRPC`, and don't need to worry if it's already a `CachingRPC`.

Reply:
maybe a branch in the AccountDBFork constructor is easier -- check if `isinstance(rpc, CachingRPC)` before wrapping it. but i think the `__new__()` technique is more robust.

Reply:
Done